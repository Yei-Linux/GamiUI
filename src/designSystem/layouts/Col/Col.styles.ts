import styled from '@emotion/styled'
import { mediaQuery } from '../../../styles/utilities/breakpoints'

export const ColWrapper = styled.div<{
  spacing: string
  xs: number
  sm: number
  md: number
  lg: number
}>`
  padding: ${(props) => props.spacing};

  flex-basis: ${(props) => `${(props.xs / 12) * 100}%`};

  ${mediaQuery('sm')} {
    flex-basis: ${(props) => `${(props.sm / 12) * 100}%`};
  }

  ${mediaQuery('md')} {
    flex-basis: ${(props) => `${(props.md / 12) * 100}%`};
  }

  ${mediaQuery('lg')} {
    flex-basis: ${(props) => `${(props.lg / 12) * 100}%`};
  }
`
