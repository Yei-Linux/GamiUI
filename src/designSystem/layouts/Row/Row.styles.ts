import styled from '@emotion/styled'

export const RowWrapper = styled.div<{
  height: string
  justifyContent: string
  alignItems: string
  isWrap?: boolean
}>`
  display: flex;
  justify-content: ${(props) => props.justifyContent};
  align-items: ${(props) => props.alignItems};
  flex-wrap: ${(props) => (props.isWrap ? 'wrap' : 'nowrap')};
  height: ${(props) => props.height};
`
