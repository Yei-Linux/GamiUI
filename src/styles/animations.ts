import { keyframes } from "@emotion/react";

export const fadeCenterIn = keyframes`
  from {
    transform: scale(.25);
    opacity: 0;
  }

  to {
    transform: scale(1);
    opacity: 1;
  }
`;

export const fadeCenterOut = keyframes`
  from {
    transform: scale(1);
    opacity: 1;
  }

  to {
    transform: scale(.25);
    opacity: 0;
  }
`;

export const fadeTopIn = keyframes`
  0% {
    -webkit-transform: translateY(-1px);
    transform: translateY(-100px);
    z-index: -1;
  }
  100% {
    -webkit-transform: translateY(0px);
    transform: translateY(0px);
    z-index: 1;
  }
`;

export const fadeTopOut = keyframes`
  0% {
    -webkit-transform: translateY(0);
    transform: translateY(0);
  }
  100% {
    -webkit-transform: translateY(-250px);
    transform: translateY(-250px);
  } 
`;