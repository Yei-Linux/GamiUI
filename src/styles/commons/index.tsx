import { css } from '@emotion/react'
import { displays } from './display'
import { flex } from './flex'

import {
  borderRadiusClasses,
  boxShadowClasses,
  fontClasses,
  opacityClasses,
  sizesClasses,
  spacingClasses,
  zIndexClasses,
} from './tokens'

export const allCommonsClasses = () => css`
  ${displays()}
  ${flex()}

  ${fontClasses()}
  ${borderRadiusClasses()}
  ${boxShadowClasses()}
  ${opacityClasses()}
  ${sizesClasses()}
  ${spacingClasses()}
  ${zIndexClasses()}
`
