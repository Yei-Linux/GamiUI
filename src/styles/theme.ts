export const colorDark = {
  primary: {
    one: "#7f9cf5",
    two: "#ffffff",
    oneLight: "rgb(24 144 255 / 20%)",
  },
  secondary: {
    one: "#ffc75f",
    two: "#94ebcd",
    three: "#374a54",
  },
  neutral: {
    one: "#000000",
    two: "#454545",
    three: "#686868",
    four: "#A4A4A4",
    five: "#D9D9D9",
    six: "#EAEAEA",
    seven: "#F2F2F2",
    eight: "#F8F8F8",
    nine: "#FFFFFF",
  },
  semantic: {
    success: "#16c79a",
    warning: "#ffd66b",
    danger: "#ff4646",
    info: "#a7c5eb",
  },
  extended: {
    code: "#374a54",
    codeStrong: "#0e141e",
    sea: "#8ac4d0",
    seaStrong: "#0278ae",
    ocean: "#b8b5ff",
    oceanStrong: "#7868e6",
    gray: "#E5E5E5",
  },
  button: {
    primary: {
      bg: "#7f9cf5",
      color: "white",
      border: "none",
    },
    secondary: {
      bg: "#ffffff",
      color: "#7f9cf5",
      border: "2px solid #7f9cf5",
    },
    tertiary: {
      bg: "#374a54",
      color: "white",
      border: "none",
    },
    cuaternary: {
      bg: "#ffffff",
      color: "#374a54",
      border: "2px solid #374a54",
    },
    text: {
      bg: "transparent",
      color: "#454545",
      border: "none",
    },
    link: {
      bg: "transparent",
      color: "#454545",
      border: "none",
    },
    success: {
      bg: "#ffffff",
      color: "#16c79a",
      border: "2px solid #16c79a",
    },
    warning: {
      bg: "#ffffff",
      color: "#ffd66b",
      border: "2px solid #ffd66b",
    },
    danger: {
      bg: "#ffffff",
      color: "#ff4646",
      border: "2px solid #ff4646",
    },
    info: {
      bg: "#ffffff",
      color: "#a7c5eb",
      border: "2px solid #a7c5eb",
    },
  },
};

export const colorLight = {
  primary: {
    one: "#7f9cf5",
    two: "#ffffff",
    oneLight: "rgb(24 144 255 / 20%)",
  },
  secondary: {
    one: "#ffc75f",
    two: "#94ebcd",
    three: "#374a54",
  },
  neutral: {
    one: "#000000",
    two: "#454545",
    three: "#686868",
    four: "#A4A4A4",
    five: "#D9D9D9",
    six: "#EAEAEA",
    seven: "#F2F2F2",
    eight: "#F8F8F8",
    nine: "#FFFFFF",
  },
  semantic: {
    success: "#16c79a",
    warning: "#ffd66b",
    danger: "#ff4646",
    info: "#a7c5eb",
  },
  extended: {
    code: "#374a54",
    codeStrong: "#0e141e",
    sea: "#8ac4d0",
    seaStrong: "#0278ae",
    ocean: "#b8b5ff",
    oceanStrong: "#7868e6",
    gray: "#E5E5E5",
  },
  button: {
    primary: {
      bg: "#7f9cf5",
      color: "white",
      border: "none",
    },
    secondary: {
      bg: "#ffffff",
      color: "#7f9cf5",
      border: "2px solid #7f9cf5",
    },
    tertiary: {
      bg: "#374a54",
      color: "white",
      border: "none",
    },
    cuaternary: {
      bg: "#ffffff",
      color: "#374a54",
      border: "2px solid #374a54",
    },
    text: {
      bg: "transparent",
      color: "#454545",
      border: "none",
    },
    link: {
      bg: "transparent",
      color: "#454545",
      border: "none",
    },
    success: {
      bg: "#ffffff",
      color: "#16c79a",
      border: "2px solid #16c79a",
    },
    warning: {
      bg: "#ffffff",
      color: "#ffd66b",
      border: "2px solid #ffd66b",
    },
    danger: {
      bg: "#ffffff",
      color: "#ff4646",
      border: "2px solid #ff4646",
    },
    info: {
      bg: "#ffffff",
      color: "#a7c5eb",
      border: "2px solid #a7c5eb",
    },
  },
};

export const palette = {
  dark: {
    color: colorDark,
  },
  light: {
    color: colorLight,
  },
};

export const spacing = {
  padding: {
    small: 10,
    medium: 20,
    large: 30,
  },
  borderRadius: {
    small: 5,
    default: 10,
  },
};

export const typography = {
  type: {
    primary: "Patrick Hand, cursive",
    secondary: "Avotte",
    terciary: "Rude1",
  },
  weight: {
    regular: "400",
    bold: "700",
    extrabold: "800",
    black: "900",
  },
  size: {
    s1: 12,
    s2: 14,
    s3: 16,
    m1: 20,
    m2: 24,
    m3: 28,
    l1: 32,
    l2: 40,
    l3: 48,
    code: 90,
  },
} as const;
