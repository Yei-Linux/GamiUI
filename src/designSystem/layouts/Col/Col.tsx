import React from 'react'
import { ColWrapper } from './Col.styles'

const spacingValues = {
  sm: '8px',
  md: '16px',
  lg: '24px',
  none: '0px'
}

type Cols = 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12

export interface ICol {
  children: React.ReactNode
  spacing: 'none' | 'sm' | 'md' | 'lg'
  xs: Cols
  sm: Cols
  md: Cols
  lg: Cols
}

const Col = ({ children, spacing, xs, sm, md, lg }: ICol) => {
  return (
    <ColWrapper
      xs={xs}
      sm={sm}
      md={md}
      lg={lg}
      spacing={spacingValues[spacing]}
    >
      {children}
    </ColWrapper>
  )
}

Col.defaultProps = {
  xs: 12,
  sm: 6,
  md: 4,
  lg: 3,
}

export default Col
