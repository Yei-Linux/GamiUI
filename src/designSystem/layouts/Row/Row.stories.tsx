import React from 'react'
import { Meta } from '@storybook/react'

import Row from '.'
import Col from '../Col'
import { colorLight } from '../../../styles/theme'

export default {
  title: 'Layout/Row',
  component: Row,
  args: { ...Row.defaultProps },
  argTypes: {},
} as Meta

export const Basic = (args: any) => (
  <Row {...args}>
    <Col spacing="sm">
      <p
        style={{
          background: colorLight.primary.one,
          color: colorLight.neutral.nine,
          padding: '16px',
        }}
      >
        1
      </p>
    </Col>

    <Col spacing="sm">
      <p
        style={{
          background: colorLight.primary.one,
          color: colorLight.neutral.nine,
          padding: '16px',
        }}
      >
        2
      </p>
    </Col>

    <Col spacing="sm">
      <p
        style={{
          background: colorLight.primary.one,
          color: colorLight.neutral.nine,
          padding: '16px',
        }}
      >
        3
      </p>
    </Col>

    <Col spacing="sm">
      <p
        style={{
          background: colorLight.primary.one,
          color: colorLight.neutral.nine,
          padding: '16px',
        }}
      >
        4
      </p>
    </Col>
  </Row>
)
Basic.args = {}
