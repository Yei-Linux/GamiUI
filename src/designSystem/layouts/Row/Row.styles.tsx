import styled from '@emotion/styled'
import { mixinFlexVariants } from 'styles/mixins/flex'
import { TAlignItems, TJustifyContent } from './Row'

export const Row = styled.div<{
  $width?: string
  $height: string
  $justifyContent: TJustifyContent
  $alignItems: TAlignItems
  $flexDirection?: string
}>`
  ${({ $justifyContent, $alignItems }) =>
    mixinFlexVariants({
      justifyContent: $justifyContent,
      alignItems: $alignItems,
    })}

  &.wrap {
    flex-wrap: wrap;
  }

  &.nowrap {
    flex-wrap: nowrap;
  }

  flex-direction: ${({ $flexDirection }) => $flexDirection};

  width: ${({ $width }) => $width};
  height: ${({ $height }) => $height};
`
