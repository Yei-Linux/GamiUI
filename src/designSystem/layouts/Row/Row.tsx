import React, { CSSProperties } from 'react'
import { RowWrapper } from './Row.styles'

export interface IRow {
  style?: CSSProperties
  children: React.ReactNode
  justifyContent:
    | 'center'
    | 'flex-start'
    | 'flex-end'
    | 'space-between'
    | 'space-around'
  alignItems: 'center' | 'flex-start' | 'flex-end' | 'normal'
  height?: string
  isWrap?: boolean
}

const Row = ({
  children,
  justifyContent,
  alignItems,
  style = {},
  height = '100%',
  isWrap = true,
}: IRow) => {
  return (
    <RowWrapper
      height={height}
      justifyContent={justifyContent}
      alignItems={alignItems}
      style={style}
      isWrap={isWrap}
    >
      {children}
    </RowWrapper>
  )
}

Row.defaultProps = {
  justifyContent: 'center',
  alignItems: 'center',
}

export default Row
