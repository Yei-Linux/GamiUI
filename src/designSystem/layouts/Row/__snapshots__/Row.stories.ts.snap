// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Layout/Row Basic 1`] = `
<div
  className="css-1e6xljk-RowWrapper e1y7zrl70"
  height="100%"
  style={Object {}}
>
  <div
    className="css-1b5jsj2-ColWrapper e17m19uc0"
    spacing="8px"
  >
    <p
      style={
        Object {
          "background": "#7f9cf5",
          "color": "#FFFFFF",
          "padding": "16px",
        }
      }
    >
      1
    </p>
  </div>
  <div
    className="css-1b5jsj2-ColWrapper e17m19uc0"
    spacing="8px"
  >
    <p
      style={
        Object {
          "background": "#7f9cf5",
          "color": "#FFFFFF",
          "padding": "16px",
        }
      }
    >
      2
    </p>
  </div>
  <div
    className="css-1b5jsj2-ColWrapper e17m19uc0"
    spacing="8px"
  >
    <p
      style={
        Object {
          "background": "#7f9cf5",
          "color": "#FFFFFF",
          "padding": "16px",
        }
      }
    >
      3
    </p>
  </div>
  <div
    className="css-1b5jsj2-ColWrapper e17m19uc0"
    spacing="8px"
  >
    <p
      style={
        Object {
          "background": "#7f9cf5",
          "color": "#FFFFFF",
          "padding": "16px",
        }
      }
    >
      4
    </p>
  </div>
</div>
`;
